### PR TITLE
 Reduce project structure & typing complexity

### LINKED ISSUE #288: Optimize web component  & page structure
### Description

As the codebase grows, it's important that we properly maintain a folder & component structure that allows us to scale. For this reason I propose the following;

* Rename every kebab-cased React Component file to Pascal-case
* Convert every React Component that is not already a folder, to a folder
* Update web-related docs on how to write new components & pages, according to the changes above

This will allow:

* Cleaner implementation
* Quicker IDE searches & indexing
* Quicker manual searching from external contributors

### Acceptance Criteria

**Every item** present under the `/src/components/` and `src/pages/` folder should be a folder with the following rules:

* The folder should be the name of the component
* The folder should contain at least one file with the exact same name (as both the folder and the component)
* The folder should contain exactly one `index.tsx` file that exports the component, along with anything else that should **publicly** be exported. No `index` file should have React code.

Example:

Let's say we have a component under `/src/components/my-component.tsx`. We would need to:

- Rename `my-component.tsx` to `MyComponent.tsx` (exactly the same name as the related react component)
- Create the folder: `/src/components/MyComponent/`
- Move the file to `/src/components/MyComponent/MyComponent.tsx`
- Add an index file to export it: `/src/components/MyComponent/index.tsx` with a code like: `export { default } from './MyComponent.tsx`. If we needed to export anything extra, then we would also add it in the `index.tsx` file.

### PR DESCRIPTION
**Pull request by alexmylonas**
## Background

Re-structuring for almost all files and folders in web folder

Closes #288,#289

## Changes

No logic change, renaming and path changes for most of the files

## Testing

Locally and deployed

### CODE REVIEW

File: deployments/core/admin_api.yml [hunk 1]
```diff
@@ -66,7 +66,7 @@ Resources:
         Variables:
           DEBUG: !Ref Debug
           USER_POOL_ID: !Ref UserPoolId
-      FunctionName: panther-users-api
+      FunctionName: panther-Users-api
```
Comment (kostaspap) [inline deployments/core/admin_api.yml]: This is not correct. This change should be reverted. I guess it slipped after some mass content replacement?

File: deployments/core/appsync.yml [hunk 2]
```diff
@@ -115,7 +115,7 @@ Resources:
       Type: AWS_LAMBDA
       ServiceRoleArn: !GetAtt AppsyncServiceRole.Arn
       LambdaConfig:
-        LambdaFunctionArn: !Sub arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:function:panther-users-api
+        LambdaFunctionArn: !Sub arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:function:panther-Users-api
```
Comment (kostaspap) [inline deployments/core/appsync.yml]: Same here.

File: deployments/core/cognito.yml [hunk 3]
```diff
@@ -71,7 +71,7 @@ Resources:
           Name: family_name
       UsernameAttributes:
         - email
-      UserPoolName: panther-users
+      UserPoolName: panther-Users
```
Comment (kostaspap) [inline deployments/core/cognito.yml]: same here

File: deployments/core/outputs.yml [hunk 4]
```diff
@@ -100,11 +100,11 @@ Resources:
         SSEEnabled: True
       TableName: panther-outputs
       # <cfndoc>
-      # This table describes the user configured destinations for alerts.
+      # This table describes the user configured Destinations for alerts.
```
Comment (kostaspap) [inline deployments/core/outputs.yml]: Should stay lowercase I think?

**Comment by alexmylonas** [comment]
@kostaspap Thanks for the catch just reverted all changes in /api & /deployments folder

File: web/src/pages/CreateLogSource/CreateLogSource.tsx [hunk 5]
```diff
@@ -20,19 +20,18 @@
 import React from 'react';
 import { Card, Flex, Alert, Box } from 'pouncejs';
 import { INTEGRATION_TYPES, AWS_ACCOUNT_ID_REGEX } from 'Source/constants';
-import Wizard from 'Components/wizard';
 import urls from 'Source/urls';
 import { extractErrorMessage } from 'Helpers/utils';
 import { useMutation, gql } from '@apollo/client';
-import { LIST_LOG_SOURCES } from 'Pages/list-log-sources/subcomponents/log-source-table';
+import { LIST_LOG_SOURCES } from 'Pages/ListLogSources/LogSourceTable';
```
Comment (3nvi) [inline web/src/pages/CreateLogSource/CreateLogSource.tsx]: import should come from `Pages/ListLogSources`

File: web/src/pages/Destinations/Columns/index.tsx [hunk 6]
```diff
@@ -0,0 +1 @@
+export { default } from './Columns';
```
Comment (3nvi) [inline web/src/pages/Destinations/Columns/index.tsx]: again columns is not a react component it should be a plain lowercased file

File: web/src/pages/Destinations/Destinations.tsx [hunk 7]
```diff
@@ -99,7 +99,7 @@ const ListDestinations = () => {
         title="Couldn't load your available destinations"
         description={
           extractErrorMessage(error) ||
-          'There was an error while attempting to list your destinations'
+          'There was an error while attempting to list your Destinations'
```
Comment (3nvi) [inline web/src/pages/Destinations/Destinations.tsx]: lowercase

File: web/src/pages/ListAlerts/Columns/index.tsx [hunk 8]
```diff
@@ -0,0 +1 @@
+export { default } from './Columns';
```
Comment (3nvi) [inline web/src/pages/ListAlerts/Columns/index.tsx]: same comment about columns again

File: web/src/pages/ListComplianceSources/ComplianceSourceColumns/ComplianceSourceColumns.tsx [hunk 9]
```diff
@@ -22,7 +22,7 @@ import React from 'react';
 import { Text, TableProps, Icon, Box } from 'pouncejs';
 import { Integration } from 'Generated/schema';
 import { generateEnumerationColumn } from 'Helpers/utils';
-import ComplianceSourceTableRowOptions from './subcomponents/compliance-source-table-row-options';
+import ComplianceSourceTableRowOptions from '../ComplianceSourceTableRowOptions/ComplianceSourceTableRowOptions';
```
Comment (3nvi) [inline web/src/pages/ListComplianceSources/ComplianceSourceColumns/ComplianceSourceColumns.tsx]: import should be flattened to `../ComplianceSourceTableRowOptions`

File: web/src/pages/ListComplianceSources/ComplianceSourceTable/ComplianceSourceTable.tsx [hunk 10]
```diff
@@ -20,10 +20,10 @@ import React from 'react';
 import { useQuery, gql } from '@apollo/client';
 import { Integration } from 'Generated/schema';
 import { INTEGRATION_TYPES } from 'Source/constants';
-import TablePlaceholder from 'Components/table-placeholder';
+import TablePlaceholder from 'Components/TablePlaceholder';
 import { Alert, Table } from 'pouncejs';
 import { extractErrorMessage } from 'Helpers/utils';
-import columns from '../compliance-source-columns';
+import columns from '../ComplianceSourceColumns/ComplianceSourceColumns';
```
Comment (3nvi) [inline web/src/pages/ListComplianceSources/ComplianceSourceTable/ComplianceSourceTable.tsx]: again flatten the improt to `../ComplianceSourceColumns `.

Also columns are not a react component (again), so we need to lowercase them and not put them in a folder

File: web/src/pages/ListComplianceSources/ListComplianceSources.tsx [hunk 11]
```diff
@@ -20,9 +20,9 @@ import React from 'react';
 import { Box, Button, Flex, Icon } from 'pouncejs';
 import { Link } from 'react-router-dom';
 import urls from 'Source/urls';
-import ErrorBoundary from 'Components/error-boundary';
-import Panel from 'Components/panel';
-import ComplianceSourceTable from './subcomponents/compliance-source-table';
+import ErrorBoundary from 'Components/ErrorBoundary';
+import Panel from 'Components/Panel';
+import ComplianceSourceTable from './ComplianceSourceTable/ComplianceSourceTable';
```
Comment (3nvi) [inline web/src/pages/ListComplianceSources/ListComplianceSources.tsx]: flatten import here as well to `./ComplianceSourceTable`

File: web/src/pages/ListResources/Columns/index.tsx [hunk 12]
```diff
@@ -0,0 +1 @@
+export { default } from './Columns';
```
Comment (3nvi) [inline web/src/pages/ListResources/Columns/index.tsx]: columns again

File: web/src/pages/ListResources/ListResourcesActions/ListResourcesActions.tsx [hunk 13]
```diff
@@ -79,12 +79,12 @@ export const filters = {
 };
 
 const LIST_ACCOUNT_IDS = gql`
-  query ListAccountIds {
-    integrations(input: { integrationType: "${INTEGRATION_TYPES.AWS_INFRA}"}) {
```
Comment (3nvi) [inline web/src/pages/ListResources/ListResourcesActions/ListResourcesActions.tsx]: who does this indentation? Prettier doesn't format it like that for me :P

File: web/src/pages/ListResources/ListResourcesTable/ListResourcesTable.tsx [hunk 14]
```diff
@@ -28,7 +28,7 @@ import { generateEnumerationColumn } from 'Helpers/utils';
 import { Table } from 'pouncejs';
 import useRouter from 'Hooks/useRouter';
 import urls from 'Source/urls';
-import columns from '../columns';
+import columns from '../Columns';
```
Comment (3nvi) [inline web/src/pages/ListResources/ListResourcesTable/ListResourcesTable.tsx]: lowercase again (columns)

File: web/src/pages/PolicyDetails/Columns/index.tsx [hunk 15]
```diff
@@ -0,0 +1 @@
+export { default } from './Columns';
```
Comment (3nvi) [inline web/src/pages/PolicyDetails/Columns/index.tsx]: columns

File: web/src/pages/ResourceDetails/Columns/Columns.tsx [hunk 16]
```diff
@@ -23,8 +23,7 @@ import { Text, TableProps, Badge, Tooltip, Label } from 'pouncejs';
 import { ComplianceItem, ComplianceStatusEnum } from 'Generated/schema';
 import { capitalize } from 'Helpers/utils';
 import { SEVERITY_COLOR_MAP } from 'Source/constants';
-import RemediationButton from 'Components/table-row-remediation-button';
-import SuppressButton from 'Components/table-row-suppress-button';
+import { SuppressButton, RemediationButton } from 'Components/TableRow';
```
Comment (3nvi) [inline web/src/pages/ResourceDetails/Columns/Columns.tsx]: I don't think we should group those components. The `TableRow` grouping doesn't make sense. Perhaps we can rename the components to `SupressButton` and `RemediationButton` and put them under `Components/Buttons/` along with `SubmitButton`

File: web/src/pages/ResourceDetails/ResourceDetails.tsx [hunk 17]
```diff
@@ -37,15 +37,17 @@ import {
...
-import ErrorBoundary from 'Components/error-boundary';
+import ErrorBoundary from 'Components/ErrorBoundary';
 import { DEFAULT_SMALL_PAGE_SIZE, INTEGRATION_TYPES } from 'Source/constants';
-import ResourceDetailsTable from './subcomponents/resource-details-table';
-import ResourceDetailsInfo from './subcomponents/resource-details-info';
-import columns from './columns';
-import ResourceDetailsPageSkeleton from './skeleton';
+import ResourceDetailsTable from './ResourceDetailsTable';
+import ResourceDetailsInfo from './ResourceDetailsInfo';
+import columns from './Columns';
```
Comment (3nvi) [inline web/src/pages/ResourceDetails/ResourceDetails.tsx]: columns again

File: web/src/pages/RuleDetails/RuleDetails.tsx [hunk 18]
```diff
@@ -28,13 +28,13 @@ import {
...
-import Panel from 'Components/panel';
+import RuleDetailsInfo from 'Pages/RuleDetails/RuleDetailsInfo';
+import Panel from 'Components/Panel';
 import urls from 'Source/urls';
 import { extractErrorMessage } from 'Helpers/utils';
-import ErrorBoundary from 'Components/error-boundary';
-import columns from './columns';
-import RuleDetailsPageSkeleton from './skeleton';
+import ErrorBoundary from 'Components/ErrorBoundary';
+import columns from './Columns';
```
Comment (3nvi) [inline web/src/pages/RuleDetails/RuleDetails.tsx]: columns

File: web/src/pages/Users/ListUsersTable/ListUsersTable.tsx [hunk 19]
```diff
@@ -20,9 +20,9 @@ import React from 'react';
 import { useQuery, gql } from '@apollo/client';
 import { ListUsersResponse, User } from 'Generated/schema';
 import { Alert, Card, Table } from 'pouncejs';
-import columns from 'Pages/users/columns';
```
Comment (3nvi) [inline web/src/pages/Users/ListUsersTable/ListUsersTable.tsx]: columns

File: web/src/pages/ResourceDetails/Columns/Columns.tsx (hunk 16 above)
Comment (alexmylonas) [reply web/src/pages/ResourceDetails/Columns/Columns.tsx]: Moved files to `Components/Buttons`!

**Pull request merged**